### PR TITLE
OCM-5627 | ci: upgrade scenario test cases

### PR DESCRIPTION
**Pull request by amalykhi**
https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-63153

 

 
Test results:
https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-63152
Screenshot from 2024-02-05 12-49-06
Screenshot from 2024-02-04 15-31-57

### CODE REVIEW

**Comment by shellymiron** [comment]
/lgtm
can you show a run for it?

File: tests/e2e/cluster_upgrade_test.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+		var targetV string
+		var clusterID string
+		BeforeEach(OncePerOrdered, func() {
+			clusterID, err = CI.PrepareRHCSClusterByProfileENV()
+			Expect(err).ToNot(HaveOccurred())
+
+		})
+		Context("Author:amalykhi-Critical-OCP-63153 @OCP-63153 @amalykhi", func() {
+			It("Update a ROSA STS cluster with RHCS provider", CI.Update,
```
Comment (yuwang-RH) [inline tests/e2e/cluster_upgrade_test.go:24]: Missing checkpoint of "downgrade" cluster to lower version.
Is there any cluster status on the terraform resource state when the cluster upgrade is triggered,like 'Scheduled/Upgrading' status? 
check the upgraded version on the terraform resource state, not only via CMS response.

File: tests/e2e/cluster_upgrade_test.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+					if CON.GetEnvWithDefault(CON.WaitOperators, "false") == "true" && !profile.Private {
+						// WaitClusterOperatorsToReadyStatus will wait for cluster operators ready
+						timeout := 60
+						err = openshift.WaitForOperatorsToBeReady(CI.RHCSConnection, clusterID, timeout)
+						Expect(err).ToNot(HaveOccurred())
+					}
+				})
+		})
+		Context("Author:amalykhi-Critical-OCP-63152 @OCP-63152 @amalykhi", func() {
+			It("Upgrade ROSA STS cluster with RHCS provider", CI.Upgrade,
```
[the remark this answers is not in the document]
Comment (amalykhi) [reply tests/e2e/cluster_upgrade_test.go:62]: Operator roles updated with the cluster together

File: tests/utils/cms/upgrade.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+		time.Sleep(1 * time.Minute)
+
+	}
+	return fmt.Errorf("ERROR!Timeout after %d minutes to wait for the policy %s into status %s of cluster %s",
+		timeout, policyID, state, clusterID)
+
+}
+
+func WaitClusterUpgradeFinished(connection *client.Connection, clusterID string) error {
```
[the remark this answers is not in the document]
Comment (amalykhi) [reply tests/utils/cms/upgrade.go:66]: I see the command exits after 2 hours anyway. Even I set 3-4 hours timeout for this command
WaitForUpgradePolicyToState(connection, clusterID, policyID, CON.Completed, 4*60)
Is there some default timeout for cms command??

Also from my experience during testing this PR, I see sometimes the Upgrade is delayed for some time and started only after 1-2 hours...

**Comment by amalykhi** [comment]
> @amalykhi @shellymiron @xueli181114 Some additional common comments. You may have some discussion and conclusion before, I just confirm with you here.
> 
> * Is there any definitions for the cluster profile using for upgrade cases? If no, do we want to use an advanced clusters to make sure the upgrading testing runing on a more customized clusters or use a basic cluster on a bisic cluster? It is more like basic clusters in the PR.
> * We hardcode the major_version as 4.14. What will we do when 4.14.z->4.15.z path is available？
> * I understand the 'update cluster' means uprading cluster within z stream while 'upgrade cluster' means upgrading cluster within Y stream here. Do we want the test title like this, upgrade cluster within Y/Z stream more explicit for me.
> 
> And @amalykhi Is there any other cases for the validations about upgrade clusters? Like empty upgrade version, invalid upgrade version, etc. If no, please add the checkpoint in these two cases.

There is no definition of upgrade case profile AFAIK, Does it exist for OCM/rosa CLI?
Once 4.15 is released we are going to upgrade this in the repo - Release will require changes in the openshift/release.
For example, there are 4.15 versions exist in candidate channel, but I would like to verify latest GAed version


Regarding the label name, both looks fine for me, just wanted to separate the test cases between Y and Z stream upgrades.
@xueli181114 @shellymiron WDYT?

File: tests/ci/profiles/tf_cluster_profile.yml [hunk 4]
```diff
@@ -79,14 +79,46 @@ profiles:
...
     proxy: false
     labeling: false
+    tagging: true
+    channel_group: stable
+    zones: ""
+    imdsv2: ""
+    oidc_config: "managed"
+    admin_enabled: true
+    unified_acc_role_path: "/uni-fied/"
+# rosa-up-z :: creating a cluster for y-stream upgrade purpose
```
Comment (shellymiron) [inline tests/ci/profiles/tf_cluster_profile.yml:94]:
```suggestion
# rosa-up-z :: creating a cluster for z-stream upgrade purpose
```

**Pull request merged**